kernel/src/table_changes/log_replay.rs

```diff
@@ -52,251 +55,258 @@ pub(crate) fn table_changes_action_iter(
     commit_files: impl IntoIterator<Item = ParsedLogPath>,
     table_schema: SchemaRef,
     physical_predicate: Option<(ExpressionRef, SchemaRef)>,
+    mut table_configuration: TableConfiguration,
```

Comment:
I _suppose_ this is okay since we are taking ownership of the `TableConfiguration` and just updating it as we go.. Though I wonder if there would be a better (read: less mutability) way to do updates to configuration leveraging `Cow` and some ability to construct new TableConfiguration incrementally? I don't have a proposal here, just that we may want to revisit this

Reply:
Cow prolly doesn't make sense in this scenario since I think that works better for pointer-based data (like graphs, linked lists, trees, etc).  I think immutability is gonna take some funky iterator stuff, but I can revisit this in the future.